lib/browser/platform/ios.rb

```diff
@@ -3,10 +3,21 @@ module Browser
   class Platform
     class IOS < Base
       MATCHER = /(iPhone|iPad|iPod)/
-      VERSION_MATCHER = /OS ([\d.]+)/
+      VERSION_MATCHER = /OS ((?<major>\d)_(?<minor>\d)_?(?<patch>\d)?)/
 
       def version
-        ua[VERSION_MATCHER, 1] || "0"
+        matches = VERSION_MATCHER.match(ua)
```

Comment:
So, I can get rid of the extraneous `if` and return early, but I think we do need to still check that the version isn't "0".  What do you think? It would break backwards compatibility and you'd need to release a major version otherwise, no?